agent/agent.go

```diff
@@ -758,6 +765,13 @@ func (a *agent) run(ctx context.Context) error {
 			} else {
 				a.setLifecycle(ctx, codersdk.WorkspaceAgentLifecycleReady)
 			}
+
+			dur := time.Since(start).Seconds()
```

Comment:
Should we stop the timer earlier, before the `if err != nil` block? Somebody may accidentally introduce a long-running operation in the future.